.coverage.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+source "${ICI_SRC_PATH}/workspace.sh"
+source "${ICI_SRC_PATH}/util.sh"
+
+echo "Checking coverage for [$COVERAGE_PKGS]"
+coverage_pass=true
+
+for pkg in $COVERAGE_PKGS; do
+    echo "Creating coverage for [$pkg]"
+    ws=~/target_ws
+    extend="/opt/ros/$ROS_DISTRO"
+    ici_exec_in_workspace "$extend" "$ws" catkin build $pkg -v --no-deps --catkin-make-args ${pkg}_coverage
+    cd $TARGET_REPO_PATH
+
+    if [ -a $ws/build/$pkg/${pkg}_coverage.info.cleaned ]; then
+        echo "Coverage summary for $pkg ----------------------"
+        lcov --summary $ws/build/$pkg/${pkg}_coverage.info.cleaned
+        echo "---------------------------------------------------"
+
+        line_cov_percentage=$(lcov --summary $ws/build/$pkg/${pkg}_coverage.info.cleaned 2>&1 | grep -Poi "lines\.*: \K[0-9.]*")
+    else
```

Comment:
I admittedly have not checked in real life what you do here. But just humor me:  What happens if I have a package without C++ or Python code or more generally an empty package with empty src folder? Do I get an error or something, even though I have no code to check?

Reply:
You will get an error if it is included here:

>       DOCKER_RUN_OPTS='-e COVERAGE_PKGS="pilz_extensions pilz_robot_programming"'

Reply:
Is this the behavior we want?

Reply:
Yes. If you have no code to check, you should not include the package like above.